### ISSUE TITLE
508-defect-1 [SCREENREADER]: STEM Standalone - Screen reader users are required to enter direct deposit details before informing them it is optional

### ISSUE THREAD

**Issue by noahgelman** [department-of-veterans-affairs/va.gov-team#18256]
# [508-defect-1](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-1)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

Noah Gelman

## User Story or Problem Statement

Screen reader users are required to enter direct deposit details before informing them it is optional

## Details

"I don't want to use direct deposit" checkbox is displayed after requiring users to enter their information.

## Acceptance Criteria

- [ ] Move "I don't want to use direct deposit" checkbox above the fields that require user to type in their direct deposit information

## Steps to Recreate

https://staging.va.gov/education/other-va-education-benefits/stem-scholarship/apply-for-scholarship-form-22-10203/applicant/information

Navigate to Step 5 of 6: Personal information

**Comment by desireeturner** [comment]
@amyknox @JenLJ @watersemma When you all have a chance can you review this 508 issue and provide feedback or sign off on the proposed solution described? Thank you!

**Comment by dneel-bah** [comment]
@noahgelman This matches the pattern outlined here: https://design.va.gov/patterns/form-templates#direct-deposit

Does this issue need to be passed over to platform, or should it be closed out?

**Comment by noahgelman** [comment]
@dneel-bah It would need to passed over to them. There is an accessibility issue related to the pattern that needs to be corrected

**Comment by af-manish** [comment]
@noahgelman <cc @ERRonat0609 @mrblanco1 > gentle nudge here

**Comment by ERRonat0609** [comment]
@af-manish @mrblanco1 - If, as of 9/15, @noahgelman stated that this needs to be turned over to Platform because it conforms to the pattern in the platform design specs and there is an accessibility issue, why are we asking for additional information? Shouldn't we turn this over to Platform?

**Comment by ERRonat0609** [comment]
@rtwell - This 508 issue was determined to be a problem with the design pattern itself, here:  https://design.va.gov/patterns/form-patterns#direct-deposit. 

I will be filling out a bug report for this pattern, which is what I was instructed to do so that this can be addressed by platform design. However, the question is still whether or not we should attempt to change this on the form ourselves or whether we need to wait until the 508 issue is officially addressed in the pattern and then re-implement the pattern on the form?

Please advise how we should proceed.

**Comment by rtwell** [comment]
let me check with the Platform folks—I'll get back asap.

**Comment by rtwell** [comment]
the design system team will tackle this issue—the STEM team does not have to fix this

**Comment by k80bowman** [comment]
Putting this one in the design system team queue.

**Comment by ERRonat0609** [comment]
@rtwell - OK, sounds like a plan. I will remove the AFS labels from this story. So do I need to still submit a bug report or just leave it at this ticket?

**Comment by k80bowman** [comment]
We are happy to track the work with this ticket, no need to submit an additional bug report.

**Comment by rtwell** [comment]
thank you all! :high-five:

**Comment by humancompanion-usds** [comment]
While we do document the patterns, the Design System Team does not code the patterns. I believe the Profile team should make this adjustment if they see fit. @ajakabcin - Is this something the Auth Exp. can take onto your backlog? We're happy to document the result of making a change to this pattern. But we need a team to try it and test it before we can document it.

**Comment by joshkimux** [comment]
@ajakabcin @andaleliz adding on the profile label to this ❤️

**Comment by ajakabcin** [comment]
thanks @joshkimux - tagging the new PM for profile :) @mtcA6

**Comment by andaleliz** [comment]
I think this is work for the team who manages the STEM application form; the steps to recreate don't reference the VA.gov profile. Unless I'm missing something?

**Comment by joshkimux** [comment]
@andaleliz 100% on me for not adding context here, I should have edited the original ticket
In an earlier comment, @rtwell noted that this is in relation to the [direct deposit pattern](https://design.va.gov/patterns/ask-users-for/direct-deposit). I don't have much more context than that, but maybe @humancompanion-usds could chime in here.

**Comment by andaleliz** [comment]
Thanks for pointing me to that @joshkimux! Our profile product only uses the direct deposit card, and doesn't include the [checkbox in question for this ticket](https://design.va.gov/patterns/ask-users-for/direct-deposit#i-dont-want-to-use-direct-deposit-checkbox). I don't think this impacts the profile, but it's totally possible I'm still missing something! 

Based on my understanding, there are two action items for this ticket:
1. [the pattern documentation needs to be updated](https://design.va.gov/patterns/ask-users-for/direct-deposit#i-dont-want-to-use-direct-deposit-checkbox) to reflect the a11y guidelines.  Or maybe reconsidered all together - what if we start people with a "yes/no" "Do you want to add direct deposit information"? and Then only take them to the card as a next step if they choose yes? 🤔 
2. the STEM form (and any others that use this pattern) needs to be updated to match the guidance once it's updated.

**Comment by joshkimux** [comment]
@andaleliz that's super context. This may then be a platform design system ticket to take on given it's not in scope with what profile is currently using. I can edit the original ticket to reflect this ❤️

**Comment by andaleliz** [comment]
>This may then be a platform design system ticket to take on given

But, also the STEM team. I can make that ticket if it's helpful?

**Comment by joshkimux** [comment]
@andaleliz agree! I've written that into the success criteria in the above ticket so (a) it doesn't get lost and (b) we can hold off on writing the ticket for the STEM team until a final recommendation is selected ✨

**Comment by humancompanion-usds** [comment]
Okay. Couple of things. 

1. We do indeed collect Direct Deposit information in the Profile. I've now added that as an example to the pattern (PR in flight). However, when I last commented here I erroneously thought that Profile handled pre-fill from Profile. Turns out the Forms library manages that piece. That code base is now my albatross to deal with.
2. Pre-fill - Pre-fill from Profile is in very limited use. We've discovered on the Veteran-facing Forms team that it largely doesn't work in the Forms library in an easily reproducible way and thus doesn't get used. Thus it is, for the most part, aspirational. As such, I've removed that documentation from the pattern entirely. It's definitely not implemented any longer in the Rogers STEM application nor in the Education Benefits (form 1990) application which I confirmed just now in staging. 

This is a great example of writing a pattern before there is an actual pattern.

**Comment by humancompanion-usds** [comment]
Since this issue was filed against the pattern [and the pattern has been updated](https://design.va.gov/patterns/ask-users-for/direct-deposit), I'm closing this. However, I believe the issue still exists in the mock-form in the Forms library. Thankfully no form is using that default version.
